src/main/java/seedu/address/logic/commands/exceptions/BuyCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.logic.commands.exceptions;
+
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXPIRY_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GOOD_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_QUANTITY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TRANSACTION_DATE;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.good.Good;
+import seedu.address.model.good.GoodName;
+import seedu.address.model.good.GoodQuantity;
+
+/**
+ * Buys the stated quantity of the good stated from the given supplier on the provided transaction date.
+ * This batch of goods will be set to expire on the given expiry date, after which they will expire and
+ * be removed from inventory.
+ */
+public class BuyCommand extends Command {
+    public static final String COMMAND_WORD = "buy";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Buys the stated "
+            + "quantity of the good stated from the given supplier on the provided transaction date. "
+            + "This batch of goods will be set to expire on the given expiry date, after which "
+            + "they will expire and be removed from inventory. "
+            + "Parameters: "
+            + PREFIX_NAME + "SUPPLIER_NAME "
+            + PREFIX_GOOD_NAME + "GOOD_NAME "
+            + PREFIX_QUANTITY + "QUANTITY "
+            + PREFIX_EXPIRY_DATE + "EXPIRY_DATE "
+            + PREFIX_TRANSACTION_DATE + "TRANSACTION_DATE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "Leong Fai Produce Co. "
+            + PREFIX_GOOD_NAME + "Apples "
+            + PREFIX_QUANTITY + "50 "
+            + PREFIX_EXPIRY_DATE + "2020-02-02 "
+            + PREFIX_TRANSACTION_DATE + "2020-01-01";
+
+    public static final String MESSAGE_SUCCESS = "Bought %1$d %2$ss";
```

Comment:
Might want to consider renaming this to SUCCESS_MESSAGE_FORMAT?

Reply:
I was following the same naming convention we inherited from AB3. MESSAGE_SUCCESS is used in add and clear commands. Should we review this convention?